File: src/backend/routes/riskMap.js
const express = require('express');
const { generateRiskMap } = require('../controllers/riskMapController');

const router = express.Router();

/**
 * @swagger
 * tags:
 *   name: Risk Map
 *   description: Geo-spatial risk mapping
 */

/**
 * @swagger
 * /risk-map:
 *   post:
 *     summary: Generate geo-spatial risk maps
 *     tags: [Risk Map]
 *     requestBody:
 *       required: true
 *       content:
 *         application/json:
 *           schema:
 *             type: object
 *             properties:
 *               spatialData:
 *                 type: array
 *                 items:
 *                   type: object
 *                   properties:
 *                     latitude:
 *                       type: number
 *                     longitude:
 *                       type: number
 *                     riskFactor:
 *                       type: number
 *     responses:
 *       200:
 *         description: Risk map generated successfully
 *       500:
 *         description: Failed to generate risk map
 */

// Generate geo-spatial risk maps
router.post('/', generateRiskMap);

module.exports = router;

File: src/backend/utils/risk_map.py
import json
import sys
import numpy as np
import matplotlib.pyplot as plt

# Function to simulate geo-spatial risk mapping
def generate_risk_map(data):
    # Example: Generate a heatmap based on spatial data
    latitudes = [point['latitude'] for point in data['spatialData']]
    longitudes = [point['longitude'] for point in data['spatialData']]
    risk_factors = [point['riskFactor'] for point in data['spatialData']]

    # Simulate heatmap generation (replace with actual GeoAI logic)
    plt.scatter(longitudes, latitudes, c=risk_factors, cmap='hot', s=100)
    plt.colorbar(label='Risk Factor')
    plt.xlabel('Longitude')
    plt.ylabel('Latitude')
    plt.title('Geo-spatial Risk Map')
    plt.savefig('risk_map.png')

    return {'message': 'Risk map generated successfully', 'file': 'risk_map.png'}

if __name__ == "__main__":
    # Read input data from stdin
    input_data = sys.stdin.read()
    data = json.loads(input_data)

    # Perform risk map generation
    try:
        result = generate_risk_map(data)
        print(json.dumps(result))
    except Exception as e:
        print(json.dumps({'error': str(e)}), file=sys.stderr)
        sys.exit(1)